### ISSUE TITLE
Windows: Could not Compile Sparkleshare in Visual Studio

### ISSUE THREAD

**Issue by tobitech** [hbons/SparkleShare#1623]
I'm trying to debug and compile SparkleShare in Visual Studio, when I click the Start debug button nothing happens. it starts debugging and just terminates and nothing else happen. how do i fix this issue?

From the Output:
The thread 0x1850 has exited with code 259 (0x103).
'SparkleShare.vshost.exe' (CLR v4.0.30319: SparkleShare.vshost.exe): Loaded 'C:\Users\tobi-imm\Downloads\Compressed\SparkleShare-master\SparkleShare-master\bin\SparkleShare.exe'. Symbols loaded.
'SparkleShare.vshost.exe' (CLR v4.0.30319: SparkleShare.vshost.exe): Loaded 'C:\Users\tobi-imm\Downloads\Compressed\SparkleShare-master\SparkleShare-master\bin\SparkleLib.dll'. Symbols loaded.
A first chance exception of type 'System.DllNotFoundException' occurred in SparkleLib.dll
A first chance exception of type 'System.DllNotFoundException' occurred in SparkleLib.dll
'SparkleShare.vshost.exe' (CLR v4.0.30319: SparkleShare.vshost.exe): Loaded 'C:\Users\tobi-imm\Downloads\Compressed\SparkleShare-master\SparkleShare-master\bin\SparkleLib.Git.dll'. Symbols loaded.
A first chance exception of type 'System.ComponentModel.Win32Exception' occurred in System.dll
A first chance exception of type 'System.DllNotFoundException' occurred in SparkleLib.dll
A first chance exception of type 'System.DllNotFoundException' occurred in SparkleLib.dll
10:29:00 | Wrote crash report to C:\Users\tobi-imm\SparkleShare\crash_report.txt
The program '[6092] SparkleShare.vshost.exe: Program Trace' has exited with code 0 (0x0).
The program '[6092] SparkleShare.vshost.exe' has exited with code -1 (0xffffffff).

What I really want to do:
I also want to modify the code so that sparkleshare does not asked for server url, but to have a constant server url link whenever it is opened

**Comment by tobitech** [comment]
from the crash_report.txt i got this
SparkleShare version: 1.5.0
Operating system:     Win32NT (Microsoft Windows NT 6.2.9200.0)
System.ComponentModel.Win32Exception: The system cannot find the file specified
   at System.Diagnostics.Process.StartWithCreateProcess(ProcessStartInfo startInfo)
   at SparkleShare.SparkleController.StartSSH() in c:\Users\tobi-imm\Downloads\Compressed\SparkleShare-master\SparkleShare-master\SparkleShare\Windows\SparkleController.cs:line 234
   at SparkleShare.SparkleController.Initialize() in c:\Users\tobi-imm\Downloads\Compressed\SparkleShare-master\SparkleShare-master\SparkleShare\Windows\SparkleController.cs:line 70
   at SparkleShare.Program.Main(String[] args) in c:\Users\tobi-imm\Downloads\Compressed\SparkleShare-master\SparkleShare-master\SparkleShare\Program.cs:line 73
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()


so who knows what I should do?

**Comment by BarryThePenguin** [comment]
You need to follow the [build instructions for windows](https://github.com/hbons/SparkleShare/blob/master/SparkleShare/Windows/README.md). You need to install msysgit

**Comment by hbons** [comment]
yep, seems that's what's going on.

**Comment by tobitech** [comment]
Alright, a new challenge
After building by following the instructions (though it seemed the instructions does not apply in terms of folders and directories), now the Visual studio compiler is telling me this: 
Error	11	Could not copy "obj\Debug\SparkleLib.Git.dll" to "..\..\..\..\bin\SparkleLib.Git.dll". Exceeded retry count of 10. Failed.	SparkleLib.Git

**Comment by tobitech** [comment]
Why did you close the issue, there are some issues still to deal with Hylke Bons

**Comment by BarryThePenguin** [comment]
Try deleting all your bin and obj folders

**Comment by tobitech** [comment]
Okay, It compiled, but I need technical guidance for what exactly i want to use the source code for which is I want to make it available for other users to connect to a private cloud server like dropbox or OneDrive, when the users start the application, they  should not have to enter the server url or host, they should just like login and start managing their own file storage i.e. syncing files on server with their local machine

**Comment by aggsol** [comment]
They will need to know where to connect to. An URL like `ssh://[user]@[host]:[port][path]` is already the easiest way to point people to a sparkle share.

**Comment by hbons** [comment]
@tobitech this tracker is for fixing SparkleShare bugs. i doubt anyone here has the time here to give you technical guidance. that said, it shouldn't be too difficult to find the relevant place in the source code, SparkleShare is not a very large project. a plugin could also help (see the wiki).

**Comment by karim10** [comment]
Hi everyone,
I have the same problem while trying to build SparkleShare from source using VisualStudio. There are many differences in my case.The first one that I have already the folder /bin in C:\Users\Me\Source\Repos\SparkleShare-1 that contains SparkleShare.exe. And when I'm trying to install msysgit (msysGit-netinstall-1.9.5-preview20150319.exe) from this [link] (https://github.com/msysgit/msysgit/releases) it crashes at the end it stops at "make[2] trying to enter directory /git".I tried to install it from this [link] (http://uncod.in/blog/installing-msysgit-on-windows7) it seems like simply installing Git and I copied C:/programs/Git to the path indicated in [the build instructions for windows](https://github.com/hbons/SparkleShare/blob/master/SparkleShare/Windows/README.md) after deleting what contains /bin folder.The build in cmd succeed and I got another Sparkleshare.exe in /bin but it craches when I try to execute it,and I have the same result of @tobitech in VisualStudio.
